File: BI-SI1/semestral/RestAPI/src/main/kotlin/cz/cvut/fit/caltrack/data/entity/Food.kt
package cz.cvut.fit.caltrack.data.entity

import javax.persistence.*

@Entity
data class Food (
    @Column(nullable = false) val name: String,
    @Column(nullable = false) val unit: String,
    @Column(name="unit_to_grams", nullable = false) val unitToGrams: Double,
    @Column(nullable = false) val fat: Int,
    @Column(nullable = false) val carbohydrates: Int,
    @Column(nullable = false) val protein: Int,
    override val id: Int = 0
) : IEntity(id)

File: BI-SI1/semestral/Aplikace/CalTrack/Model/Food.swift
//
//  Food.swift
//  CalTrack
//
//  Created by Ondřej Wrzecionko on 19.04.2021.
//

import Foundation

struct Food : Codable, Identifiable {
    let id: Int
    let name: String
    let unit: String
    let unitToGrams: Double
    let fat: Int
    let carbohydrates: Int
    let protein: Int
}

extension Food {
    static let preview = Food(id: 1, name: "Kebab", unit: "ks", unitToGrams: 300, fat: 12, carbohydrates: 20, protein: 5)
}
